packages/web3-shared/base/src/helpers/formatCurrency.ts

```diff
@@ -48,28 +50,46 @@ const formatCurrencySymbol = (symbol: string, isLead: boolean) => {
     return isLead || symbol.length === 0 ? symbol : ` ${symbol}`
 }
 
+const fiatCurrencyResultModifier = (
+    result: string,
+    currency: LiteralUnion<Keys | 'USD'> = CurrencyType.USD,
+    isAssetValue: boolean,
+) => {
+    if (currency === CurrencyType.HKD) return result.replaceAll('$', 'HK$')
+
+    if (currency === CurrencyType.JPY && isAssetValue)
+        return result.startsWith('¥') ? '¥' + Number(result.replace('¥', '')).toFixed() : result
+    return result
+}
+
 // https://mask.atlassian.net/wiki/spaces/MASK/pages/122916438/Token
 export function formatCurrency(
     inputValue: BigNumber.Value,
     currency: LiteralUnion<Keys | 'USD'> = CurrencyType.USD,
     options?: FormatterCurrencyOptions,
 ): string {
-    const { onlyRemainTwoDecimal = false, fiatCurrencyRate = 1, customDecimalConfig } = options ?? {}
+    const { isAssetValue = false, fiatCurrencyRate = 1, customDecimalConfig } = options ?? {}
```

Comment:
The original name is more precise. Name like isAssetValue is uncertain about what is the result actually looks like. By the way, as a currency formatter, there is no need to concern about who uses it.